pkg/deployment/cert.go

```diff
@@ -185,7 +185,7 @@ func GetTLSNodeCert(ctx context.Context, helper *helper.Helper,
 			Usages:      usages,
 		}
 
-		certSecret, result, err = certmanager.EnsureCert(ctx, helper, request)
+		certSecret, result, err = certmanager.EnsureCert(ctx, helper, request, instance)
```

Comment:
How do we handle scenario where multiple deployments have the same nodeset and we request certs for the nodes in the nodeset? Should we check for existing certs (are they even going to change) or just patch the certs without worrying like now[1]?

[1] https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/certmanager/certificate.go#L229

Reply:
@rabi  the current logic checks for the existence of the secret that is created by cert-manager when the certificate is issued (i.e. in line 168).  If a secret is found, then err == nil and the code in 171-193 is not invoked.

So subsequent deployments on the same nodeset should not trigger any code to patch the certs.

If a secret is not found, then we enter the loop and either error out (because of some error other than not-found , or we issue the certs).